### PR TITLE
[Resources] Add new API version

### PR DESCRIPTION
**Pull request by filizt**
This PR has the following changes for the new API version 2020-06-01:
*  `id` and `relativePath` properties added to `TemplateLink` object (this is to support Template Specs)
* `provisioningState` property in `DeploymentPropertiesExtended` is now an enum 
* `statusMessage` property in `DeploymentOperationProperties` now has a defined type 

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/202/Overall-Process-of-AME-Onboarding) for the workflow.
- [x] [Validation tools](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/84/Swagger-Validation-tools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/FAQ.md).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process]( https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review).

### CODE REVIEW

**Comment by filizt** [comment]
@raych1 PR is ready for review. Could you add WaitForARMFeedback label? 

Avocado is complaining about examples files not being referenced from the swagger but they are - could you look at this?

**Comment by raych1** [comment]
@filizt , avocado failures is caused by the example files are not referenced in the swagger spec. If the example files are not used, please remove them.

**Comment by filizt** [comment]
@Ray Chen examples files are used and they are being referenced from the swagger. I am not sure why there are errors. Could you help with that?

---
In reply to: [637354911](https://github.com/Azure/azure-rest-api-specs/pull/9487#issuecomment-637354911) [](ancestors = 637354911)

**Comment by raych1** [comment]
@ChenTanyi , can you pls have a look at SDK java failures? Is there any incorrection in readme.java.md?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json [hunk 1]
```diff
@@ -0,0 +1,5905 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/tagNames": {
+      "get": {
+        "tags": [
+          "Tags"
+        ],
+        "operationId": "Tags_List",
```
[the remark this answers is not in the document]
Comment (filizt) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:3514]: I see. Should I update all operationIds that start with Tags_*  or just this one "Tags_List"?

**Comment by pilor** [comment]
>           "description": "Denotes the state of the provisioning. Individual resource providers can add more specific states.",

this isn't accurate in this situation.  This is the provisioningState of the deployment.  Microsoft.Resources should have control over all the possible Microsoft.Resources/deployments provisioningStates

---
Refers to: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4648 in 92d5c49. [](commit_id = 92d5c49cf14f454c3d845fb714683eccaf775105, deletion_comment = False)

**Comment by pilor** [comment]
>         },

should add a new example or two showing how id and relateivePath are used in template spec scenarios

---
Refers to: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4233 in 92d5c49. [](commit_id = 92d5c49cf14f454c3d845fb714683eccaf775105, deletion_comment = False)

**Review by pilor — CHANGES_REQUESTED** [review]
:clock1:

**Comment by filizt** [comment]
>         },

Gokul do you mind adding examples scenarios with id and relativePath properties in Template Specs?

---
In reply to: [639167519](https://github.com/Azure/azure-rest-api-specs/pull/9487#issuecomment-639167519) [](ancestors = 639167519)

---
Refers to: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4233 in 92d5c49. [](commit_id = 92d5c49cf14f454c3d845fb714683eccaf775105, deletion_comment = False)

**Comment by filizt** [comment]
>           "description": "Denotes the state of the provisioning. Individual resource providers can add more specific states.",

There was a confusion over the resource provisioning state vs deployment provisioning state. Updated now.

---
In reply to: [639166394](https://github.com/Azure/azure-rest-api-specs/pull/9487#issuecomment-639166394) [](ancestors = 639166394)

---
Refers to: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4648 in 92d5c49. [](commit_id = 92d5c49cf14f454c3d845fb714683eccaf775105, deletion_comment = False)

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChenTanyi) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:3514]: All `Tags_*` need to be changed. Thanks!

**Comment by Xynoclafe** [comment]
> 
> 
> > ```
> >     },
> > ```
> 
> @Xynoclafe do you mind adding example Template Specs scenarios with id and relativePath properties?
> 
> In reply to: [639167519](https://github.com/Azure/azure-rest-api-specs/pull/9487#issuecomment-639167519) [](ancestors = 639167519)
> 
> Refers to: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4233 in [92d5c49](https://github.com/Azure/azure-rest-api-specs/commit/92d5c49cf14f454c3d845fb714683eccaf775105). [](commit_id = [92d5c49](https://github.com/Azure/azure-rest-api-specs/commit/92d5c49cf14f454c3d845fb714683eccaf775105), deletion_comment = False)

Added a couple of examples for TemplateSpecs.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (filizt) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json]: Thanks! @Grayer123  FYI for the change.

[the remark this answers is not in the document]
Comment (filizt) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json]: @ChenTanyi updated the tags however Java SDK task is still failing. Could you take a look?

**Review by pilor — APPROVED** [review]
:shipit:

**Review by raych1 — APPROVED** [review]

**Pull request merged**